### PR TITLE
[BEAM-8213] Split out lint job from monolithic python preCommit tests on jenkins

### PR DESCRIPTION
**Pull request by chadrik**
Take two.

After discovering that #9642 would increase the jenkins pre-commit backlog by ~66% we decided to focus on the aspect of that PR that was a clear winner: splitting out the lint job.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by chadrik** [comment]
R: @youngoli 
R: @tvalentyn
R: @aaltay 
R: @robertwb

File: build.gradle [hunk 1]
```diff
@@ -102,7 +102,7 @@ rat {
 
     // Json doesn't support comments.
     "**/*.json",
-      
+
```
Comment (chadrik) [inline build.gradle]: I just noticed my editor cleaned this up.  Let me know if you want me to remove it.

**Comment by chadrik** [comment]
Run CommunityMetrics PreCommit
Run Java PreCommit
Run Python_PVR_Flink PreCommit

**Comment by chadrik** [comment]
Run Seed Job

**Comment by tvalentyn** [comment]
Run Seed Job

**Comment by tvalentyn** [comment]
The change LGTM, let's re-run the tests once seed job finishes (previous seed job run was overwritten by the scheduled seed job run)

**Comment by tvalentyn** [comment]
Thanks, @chadrik !

**Comment by chadrik** [comment]
Run CommunityMetrics PreCommit
Run Java PreCommit
Run Python_PVR_Flink PreCommit

**Comment by chadrik** [comment]
Run CommunityMetrics PreCommit

**Comment by chadrik** [comment]
Run Java PreCommit

**Comment by chadrik** [comment]
Run Python_PVR_Flink PreCommit

**Comment by chadrik** [comment]
Run Python_PVR_Flink PreCommit

**Comment by tvalentyn** [comment]
Run Python_PVR_Flink PreCommit

**Comment by chadrik** [comment]
I'm not sure why these keep failing....

The build scan link doesn't work for me "Your build scan could not be displayed"

**Comment by chadrik** [comment]
I triple checked that I haven't changed anything about Python_PVR_Flink.

**Comment by tvalentyn** [comment]
Python_PVR_Flink failures are not related. Build scan is broken for a different reason (there is a thread on dev@ about that).

**Comment by tvalentyn** [comment]
Run Python PreCommit

**Comment by tvalentyn** [comment]
Running the python precommit to make sure it still works now that the seed job succeded.

**Comment by tvalentyn** [comment]
We can see that precommit was triggered successfully, I think this is ready to merge.

**Pull request merged**